SS_Activation_OpenMC.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb  8 08:16:12 2024
+
+@author: Anupama Rajendra
+"""
+import openmc
+import openmc.deplete
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+
+# Importing Vitamin-J energy group structure:
+# This excel file contains the energy bounds of the Vitamin J structure
+Vit_J = pd.read_excel('VitaminJEnergyGroupStructure.xlsx')
+ebounds = Vit_J.iloc[:, 1]
+
+openmc.config['chain_file'] = 'chain_endfb71_sfr.xml'
+
+# Create materials & export to XML:
+#Simulating tungsten shell:
+W = openmc.Material(material_id=1, name='W_Shell')
+W.set_density('g/cm3', 19.28)
+W.add_element('W', 1.00)
+materials = openmc.Materials([W])
+materials.export_to_xml()
+
+# Create geometry
+#Spherical shell:
+    
+R_1 = 1000
+S_1= openmc.Sphere(r=R_1) #sphere of radius 1000cm
+inside_sphere_1 = -S_1
+outside_sphere_1 = +S_1
+R_2 = 1005
+S_2 = openmc.Sphere(r=R_2, boundary_type='vacuum')
+inside_sphere_2 = -S_2
+outside_sphere_2 = +S_2
+S_3 = outside_sphere_1 & inside_sphere_2 #filled with tungsten
+
+# Mapping materials to geometry:
+Void = openmc.Cell(fill=None, region = inside_sphere_1)
+Shell = openmc.Cell(fill=W, region=S_3)
+geometry = openmc.Geometry([Void, Shell])
+geometry.export_to_xml()
+
+# Source distribution:
+PointSource = openmc.stats.Point(xyz=(0.0, 0.0, 0.0))
+Prob = openmc.stats.Discrete(14E+06, 1.0)
+
+# Assign simulation settings
+settings = openmc.Settings()
+settings.batches = 10
+settings.inactive = 1
+settings.particles = 10000
+settings.source = openmc.Source(space=PointSource, energy=Prob, strength = 1.0, particle = 'neutron')
+settings.run_mode = 'fixed source'
+settings.export_to_xml()
+
+# Define tallies
+neutron_tally = openmc.Tally(name="Neutron tally")
+neutron_tally.scores = ['flux', 'elastic', 'absorption']
+# Implementing filter for neutron tally through W shell
+cell_filter = openmc.CellFilter([Shell])
+neutron_tally.filters = [cell_filter]
+
+# Creating a tally to get the flux energy spectrum.
+# An energy filter is created to assign to the flux tally using the Vitamin J structure.
+energy_filter_flux = openmc.EnergyFilter(ebounds)
```

Comment:
I think I just learned that OpenMC has the 175 group Vitamin-J structure built in.

See the `from_group_structure` method [here](https://docs.openmc.org/en/stable/pythonapi/generated/openmc.EnergyFilter.html) and the library of structures [here](https://docs.openmc.org/en/v0.12.1/pythonapi/mgxs.html)

Reply:
Do you have a preference as to how the structure is implemented? (I'm guessing the built-in one)

Reply:
I think it's best to switch to the built-in groups, yes.

Reply:
Sounds good. Made this change and the results appear to be the same.